sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/AssetFileType.cs

```diff
@@ -19,13 +20,23 @@ namespace Azure.MixedReality.ObjectAnchors.Conversion
         internal const string ObjValue = ".obj";
         internal const string PlyValue = ".ply";
 
+        internal string Value => _value ?? string.Empty;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="AssetFileType"/> struct.
         /// </summary>
         /// <param name="value">The asset file type.</param>
         public AssetFileType(string value)
         {
-            _value = value ?? throw new ArgumentNullException(nameof(value));
+            Argument.AssertNotNullOrWhiteSpace(value, nameof(value));
+
+            if (!value.StartsWith(".", StringComparison.Ordinal))
+            {
+                throw new ArgumentException("The value must start with a '.' character.", nameof(value));
+            }
+
+            _value = value.ToLowerInvariant()
+                ?? throw new ArgumentNullException(nameof(value));
```

Comment:
How can toLowerInvariant return null? Otherwise, why the extra null check here?

Reply:
It can't. Sorry. I moved the validation up above and forgot to "complete" the move.